File: Python/monte-carlo/montecarlo.py

import random
import time

def monteCarlo() :
   cont = 0
   dentro=0
   fora=0
   for j in range(1000000):
      x=2*random.random()-1   
      y=2*random.random()-1  
      
      ponto = x**2 + y**2
      
      if (ponto<=1):        
         dentro +=1
         
      else:            
         fora +=1
   
   valor_pi = 4*dentro/(dentro+fora)
   return valor_pi

def main():
  
   arq = open("runtime-montecarlo.csv","w")
   media = 0.0
   
   for i in range(100):
      inicio = time.time()
      valor_pi= monteCarlo()
      fim = time.time()

      tempo = (fim-inicio)*1000 #milissegundos
      media = media+tempo
      arq.write(str(tempo)+"\n")

   media = media/100
   
   print("Tempo medio das execucoes "+str(media)+"\n")
   arq.close()

main()

File: C/monte-carlo/montecarlo.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>
#include <math.h>

int main(){

    clock_t tempo;
    srand(time(NULL));
   

    int cont=0, dentro=0, fora=0;
    int i;
    float aux1,aux2;

    float x,y,ponto,sorteio, valor_pi;
    FILE* arq;
    arq = fopen("runtime-montecarlo.txt","wt");
    //fprintf(arq,"Runtime Monte Carlo\n");   
    
    while(cont < 100){
        tempo = clock();
        for(i=0;i<1000000;i++){
         
            sorteio = (float)rand()/(float)RAND_MAX;
            
            x= 2 * sorteio -1;
            
            sorteio = (float)rand()/(float)RAND_MAX;
            y= 2 * sorteio -1;
            
            ponto = x*x + y*y;

            if(ponto<=1)
            dentro++;
            
            else
                fora++;
            
        }
        
        aux1 = 4*dentro;
        aux2 = dentro+fora;

        valor_pi = aux1 / aux2;
        
        //fprintf(arq,"Execucao  %d\n", cont);   
        //fprintf(arq,"Valor simulado de pi %f\n", valor_pi);   
        fprintf(arq,"%f\n", ((clock() - tempo) / (double)CLOCKS_PER_SEC)*1000);
        //fprintf(arq,"###################\n");      
        cont++;
    }


    fclose(arq);


    return 0;
}

